openlibrary/plugins/upstream/addtag.py

```diff
@@ -102,16 +117,19 @@ def no_match(self, i: web.utils.Storage) -> NoReturn:
         raise safe_seeother(key)
 
 
+# XXX : What is the actual point of this?
 # remove existing definitions of addtag
 delegate.pages.pop('/addtag', None)
 
 
+# XXX : Why does this redirect exist?
```

Comment:
Commenting out this class and the `/addtag` pop on line 122 and restarting `web` yields a `404` page when `/addtag` is requested:

![image](https://github.com/user-attachments/assets/0423d65d-9f5e-43c4-9bc0-a1794198b6cd)

Probably safe to remove these, then?

Reply:
Check for reference to `addtag` in the codebase.

if none, remove